book/src/04_3_i2c.md

```diff
@@ -2,16 +2,99 @@
 
 ## Introduction
 
-TODO describe protocol & purpose
-TODO https://learn.sparkfun.com/tutorials/i2c
+The I2C (Inter-Integrated Circuit) protocol allows multiple peripheral chips to communicate with one or more controller chips. Many devices can be connected to the same I2C bus and messages can be send to a particular device by specifying its I2C address. The protocol requires two signal wires and can only be used for short distance communications within one device. 
 
+One of the signal lines is for data (SDA) and the other is for the clock signal (SCL). An I2C transaction consists of one or more messages. Each message begins with a start symbol. A message is either write or read, indicated by the next bit in the row. Following this comes the actual message in form of bytes. The message ends with a stop symbol. The clock signal rises and falls in a specified frequency.
+
+TODO I2C signal image
 ## I2C Driver Exercise
 
+We're not going to write an entire driver, merely the first step: the `hello world` of driver writing: reading the device ID of the sensor. 
+
 TODO `i2c-driver/exercise/src/imc42670p.rs` is a gap text of a very basic i2c IMU sensor driver. The task is to complete the file, so that running `main.rs` will log the device ID of the driver. The this gap text driver is based on the version of the same name that lives in common, but provides a little bit more functionality.
```

Comment:
```suggestion
TODO `i2c-driver/exercise/src/imc42670p.rs` is a gap text of a very basic I²C IMU sensor driver. The task is to complete the file, so that running `main.rs` will log the device ID of the driver. The this gap text driver is based on the version of the same name that lives in common, but provides a little bit more functionality.
```